
syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.pubref.rules_protobuf.examples.helloworld";
option java_outer_classname = "HelloWorldProto";

import "helloworld/proto/common.proto";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SendMessage (Msg) returns (MsgReply) {}
  rpc RecvMessages (MsgReq) returns (Msgs) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  common.Config config = 2;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message Msg {
  string from = 1;
  string to = 2;
  string message = 3;
}

message MsgReq {
  string uid = 1;
}

message Msgs {
  repeated Msg messages = 1;
}

message MsgReply {
  int32 status = 1;
}
